### PR TITLE
Add bgfx

### PR DESCRIPTION
**Pull request by RazielXYZ**
Specify library name and version:  **bgfx/cci.20230216**

This is the bgfx library, a powerful cross-platform rendering library that can handle all common rendering backends from a common codebase.

Similarly to bimg (https://github.com/conan-io/conan-center-index/pull/15317), this requires bx source to build, as well as bimg source. I've mapped versions the same way as in the bimg package.
Other than that, the only main difference between this package and those of bx and bimg is that bgfx does support being built and used as a shared library.

Notes from bx (https://github.com/conan-io/conan-center-index/pull/13866) and bimg still apply.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/bgfx/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
+import os
+
+required_conan_version = ">=1.50.0"
```
Comment (SpaceIm) [inline recipes/bgfx/all/test_package/conanfile.py:6]:
```suggestion
```

File: recipes/bgfx/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain"
```
Comment (SpaceIm) [inline recipes/bgfx/all/test_package/conanfile.py:10]:
```suggestion
    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
```

File: recipes/bgfx/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    test_type = "explicit"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def build(self):
+        if can_run(self):
```
Comment (SpaceIm) [inline recipes/bgfx/all/test_package/conanfile.py:20]: why this branching?

File: recipes/bgfx/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
```
Comment (SpaceIm) [inline recipes/bgfx/all/test_package/conanfile.py:23]:
```suggestion
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```

File: recipes/bgfx/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(test_package LANGUAGES CXX)
+
+find_package(bx REQUIRED CONFIG)
+find_package(bimg REQUIRED CONFIG)
+find_package(bgfx REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 14 CXX_STANDARD_REQUIRED ON CXX_EXTENSIONS OFF)
```
Comment (SpaceIm) [inline recipes/bgfx/all/test_package/CMakeLists.txt:10]:
```suggestion
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_14)
```

File: recipes/bgfx/all/conanfile.py [hunk 6]
```diff
@@ -12,9 +12,6 @@
 from pathlib import Path
 import os
 
-required_conan_version = ">=1.50.0"
```
Comment (SpaceIm) [inline recipes/bgfx/all/conanfile.py]: I was suggesting to remove this in test package, to main conanfile. You know you can accept suggestions as commits?

Comment (RazielXYZ) [reply recipes/bgfx/all/conanfile.py]: Oops, I'm blind sometimes. And I actually didn't know that, found it now!

**Comment by RazielXYZ** [comment]
Any chance we can take a look at this?

**Comment by franramirez688** [comment]
This PR is failing for Conan v2 because of some missing binaries. This PR is needed https://github.com/conan-io/conan-center-index/pull/19026 to build one of them, so let's wait for it and update the branch after that to see if it's already solved.

**Comment by RazielXYZ** [comment]
I thought the v2 pipeline succeeding wasn't required (yet)? Either way, I agree that we should make everything compatible with v2 regardless. I also plan on touching up the bx, bimg and bgfx recipes again after they're all here in this initial form.

**Review by danimtb — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Nice trick with getting the dependencies sources!

**Pull request merged**